src/actions/sources.js

```diff
@@ -50,6 +50,11 @@ function newSource(source: Source) {
       source
     });
 
+    dispatch({
+      type: constants.RESTORE_TABS,
+      source
+    });
```

Comment:
Hi @jasonLaster 

These code enable the persist tabs to show up on refresh and close/open. I haven't been able to find a better way to achieve this behavior.

[Line 53 ~ 56 in `actions/sources.js`](https://github.com/devtools-html/debugger.html/pull/1547/files?diff=unified#diff-695f8e845eb7e7930a1b665650866b59R53)
This calls the code on line 72 - 81 in `reducers/sources.js`. We're using this because when a source has been downloaded and added to the SourcesTree, we need to check whether or not this source is in the persist tabs, which happens on line 72 - 81 in `reducers/sources.js`.

[Line 72 - 81 in `reducers/sources.js`](https://github.com/devtools-html/debugger.html/pull/1547/files?diff=unified#diff-701eb9ae1383078fc8b17995acdd81bbR72)
This code adds those persist tabs in the SourceTabs. Without this code, the persist tabs would not be loaded even though we're loading the tabs in the initial state. The reason is that loading the tabs in the initial state happens before those tabs could be downloaded and added to the SourcesTree, so we need to call this code when each source has been downloaded and added to the SourceTree, which happens on line 53 ~ 56 in `actions/sources.js`  

[Line 100 ~ 108 in `SourcesTree.js`](https://github.com/devtools-html/debugger.html/pull/1547/files?diff=unified#diff-9d180fae1eb61e88c4a4cf4b60ea0eafR100)
What this does is that we only load the active tab's source text when all the sources in the tree are finished loading at this given time. However, there is an issue. This code would be executed if you select a tab or open a new file.